pkg/operator/podidentity/podidentitywebhook_controller.go

```diff
@@ -334,7 +336,13 @@ func (r *staticResourceReconciler) ReconcileResources(ctx context.Context) error
 		// Set replicas=1 for deployment on single replica topology clusters
 		requestedDeployment.Spec.Replicas = pointer.Int32(1)
 	}
-	requestedDeployment.Spec.Template.Spec.Containers[0].Image = r.imagePullSpec
+
```

Comment:
Keep the imagePullSpec substitution here in the controller code to reduce duplicate code as it is required for all platforms.